### PR TITLE
Implemented Project Guidelines for SQL Projection

### PR DESCRIPTION
**Pull request by nuxeric**
So I'm really bad @ naming things so I have a file called InventoryLogic because I did not really know why accounts was called accounts @pseigo , maybe you could explain that and then I can chain the name for this specific file (: OR you could help me name it. 

I layed out the ground work for Inventory related queries and I implemented a really simple projection that queries 4 of 5 elements from a single table. I could add an additional query with drop down tables but for tonight I was going to stop here so I could do a quiz due for tmrw morning hahaha. 

The work is done similary to how peyton has set up his logic and structure for personell. The database and model logic lives in model, routes in routes 

lmk if you have any questions please take a look

### CODE REVIEW

**Comment by nuxeric** [comment]
Actually on further thought I looked on Trello and it needs a join operation, I can just work on that after this gets merged in , probably with the same Inventory entity , maybe even include a drop down menu for projection along with the join.

File: web/tsconfig.json
Comment (pseigo) [inline web/tsconfig.json:67]: JavaScript has a really weird convention to use trailing commas.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas

File: web/src/model/inventoryLogic.ts
Comment (pseigo) [inline web/src/model/inventoryLogic.ts:12]: With respect to the rubric and work done, I can see why you chose this name, but I reckon someone who hasn't looked inside this file wouldn't know what this does. You could alternatively call it `selectInventory()` and parameterize it with booleans for each attribute.

The same goes for "inventory" vs. "inventoryLogic."

Not necessary for the project though. Best just to get it done. Thought I'd mention though. :)

**Comment by pseigo** [comment]
The motivation for "accounts" is [Elixir/Phoenix contexts](https://hexdocs.pm/phoenix/contexts.html).

- "Contexts are dedicated modules that expose and group related functionality."
  - "Encapsulat[es] patterns such as data access and data validation."
  - "Decouple[s] and isolate[s] our systems into manageable, independent parts."

> For example, anytime you call Elixir's standard library, be it Logger.info/1 or Stream.map/2, you are accessing different contexts. Internally, Elixir's logger is made of multiple modules, but we never interact with those modules directly. We call the Logger module the context, exactly because **it exposes and groups all of the logging functionality.**
>
> https://hexdocs.pm/phoenix/contexts.html

> The context is an Elixir module that serves as an API boundary for the given resource. A context often holds many related resources. Therefore, if the context already exists, it will be augmented with functions for the given resource.
> 
>> Note: A resource may also be split over distinct contexts (such as Accounts.User and Payments.User).
> 
> https://hexdocs.pm/phoenix/Mix.Tasks.Phx.Gen.Context.html

The idea is you have one public API at the top level per context and make all logic inside the context private. Data types can also go in the context and be made public. What I did is create an "Accounts" context with "Personnel" nested inside, so `accounts.ts` provides a public interface to make DB operations on Personnel without being exposed to any implementation details.

> Actually on further thought I looked on Trello and it needs a join operation, I can just work on that after this gets merged in , probably with the same Inventory entity , maybe even include a drop down menu for projection along with the join.

For inventory, I might make a context called "Storage", where you could also put Storage Containers, Inventory Types, and maybe even Warehouses. If you wanted to do that, your structure would look like

```txt
├── model
│   ├── accounts
│   │   ├── permissions.ts
│   │   └── personnel.ts
│   ├── storage
│   │   └── inventory.ts
│   │── accounts.ts
│   └── storage.ts
```

Where `storage.ts` would have your special query. I called my "list all entries" function `listPersonnel()`, so you might rename yours to `listInventory()` and parameterize it with the fields you want to project on.

> 
> 
> Actually on further thought I looked on Trello and it needs a join operation, I can just work on that after this gets merged in , probably with the same Inventory entity , maybe even include a drop down menu for projection along with the join.

If you're adding a join, then you're probably doing more than just listing inventory and I'm sure you could come up with a name to describe what it's doing. :)

Hope that helps!

**Review by pseigo — CHANGES_REQUESTED** [review]
_(waiting for response to comments)_

**Review by ShaunDonn — APPROVED** [review]

File: web/tsconfig.json
Comment (nuxeric) [reply web/tsconfig.json:67]: This is JSON "JSON, however, disallows trailing commas."
, so trailing commas should not be in here. Also throws an error in my IDE

File: web/src/model/inventoryLogic.ts
Comment (nuxeric) [reply web/src/model/inventoryLogic.ts:12]: Just read your comment below, makes more sense to me I'll change some of the naming conventions! (:

File: web/tsconfig.json
Comment (pseigo) [reply web/tsconfig.json:67]: Ohh shoot good catch

Comment (pseigo) [reply web/tsconfig.json:67]: Trailing commas should just stop being a thing 😆

File: web/src/routes/mainRouter.ts
Comment (pseigo) [inline web/src/routes/mainRouter.ts:5]:
```suggestion
import InventoryRoute from "./inventory/inventoryRoute";
```

File: web/src/routes/mainRouter.ts
Comment (pseigo) [inline web/src/routes/mainRouter.ts:11]:
```suggestion
    private inventoryRoute: InventoryRoute;
```

File: web/src/routes/mainRouter.ts
Comment (pseigo) [inline web/src/routes/mainRouter.ts:17]:
```suggestion
        this.inventoryRoute = new InventoryRoute(db);
```

File: web/src/routes/inventory/inventoryRoute.ts
Comment (pseigo) [inline web/src/routes/inventory/inventoryRoute.ts:6]:
```suggestion
export default class InventoryRoute extends ApiRoute {
```

**Review by pseigo — APPROVED** [review]

**Pull request merged**
